spec/serializers/current_user_serializer_spec.rb

```diff
@@ -283,11 +283,10 @@
       )
     end
 
-    it "isn't included when navigation menu is legacy with old user menu" do
+    it "isn't included when navigation menu is legacy" do
       SiteSetting.navigation_menu = "legacy"
-      SiteSetting.enable_new_notifications_menu = false
 
-      expect(serializer.as_json[:new_personal_messages_notifications_count]).to be_nil
+      expect(serializer.as_json[:new_personal_messages_notifications_count]).to eq(1)
```

Comment:
The description of the test says "isn't included", but the expectation is `eq(1)`. We should work out what the desired behaviour is and make things consistent.

Reply:
I think the previous behavior was if the `navigation_menu` was enabled, `enable_new_notifications_menu` would be false?

So I think now that we can have the new notifications menu with the legacy navigation menu, we will need to include the `new_personal_message_notification_count`. I'll update the test description.